pyvista/plotting/utilities/sphinx_gallery.py

```diff
@@ -53,6 +53,14 @@ def html_rst(
     return images_rst
 
 
+def _process_events_before_scraping(plotter):
+    """Process events such as changing the camera or object before scraping."""
+    if plotter.iren is not None and plotter.iren.initialized:
+        if hasattr(plotter, "app") and plotter.app is not None:
+            plotter.app.processEvents()
+        plotter.update()
```

Comment:
I want to double check that this `.update()` call is necessary or if it should just be in the nest if-block

Reply:
If it were nested in the if block then it would still solve my issue but if there were an `iren` that isn't associated with `pyvistaqt` then the `update` call would still be helpful. Not sure if that's a common use case but it seems to be logically consistent with the structure of `pyvista` as I understand it.